app/models/manageiq/providers/cloud_manager/orchestration_template_runner.rb

```diff
@@ -0,0 +1,165 @@
+class ManageIQ::Providers::CloudManager::OrchestrationTemplateRunner < ::Job
+  DEFAULT_EXECUTION_TTL = 10 # minutes
+
+  # options are job table columns, including options column which is the playbook context info
+  def self.create_job(options)
+    super(name, options)
+  end
+
+  def minimize_indirect
+    @minimize_indirect = true if @minimize_indirect.nil?
+    @minimize_indirect
+  end
+
+  def current_job_timeout(_timeout_adjustment = 1)
+    @execution_ttl ||=
+      (options[:execution_ttl].present? ? options[:execution_ttl].try(:to_i) : DEFAULT_EXECUTION_TTL) * 60
+  end
+
+  def start
+    time = Time.zone.now
+    update_attributes(:started_on => time)
+    miq_task.update_attributes(:started_on => time)
+    my_signal(false, :deploy_orchestration_stack)
+  end
+
+  def update
+    time = Time.zone.now
+    update_attributes(:started_on => time)
+    miq_task.update_attributes(:started_on => time)
+    my_signal(false, :update_orchestration_stack)
+  end
+
+  def deploy_orchestration_stack
+    set_status('deploying orchestration stack')
+
+    create_options = options.except(:service_id)
+    @orchestration_stack = ManageIQ::Providers::CloudManager::OrchestrationStack.create_stack(
+      service.orchestration_manager, service.stack_name, service.orchestration_template, create_options
+    )
+    self.name = "#{name}, Orchestration Stack ID: #{@orchestration_stack.id}"
+    miq_task.update_attributes(:name => name)
+    save!
+    service.add_resource(@orchestration_stack)
```

Comment:
move this line back to service class. You can add orchestraiton_stack_id to job's options so that the service can read back the stack.